src/components/sidebar/MarkerSidebar.js

```diff
@@ -0,0 +1,157 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+
+import * as React from 'react';
+import classNames from 'classnames';
+
+import explicitConnect from '../../utils/connect';
+import { selectedThreadSelectors } from '../../reducers/profile-view';
+import { getSelectedThreadIndex } from '../../reducers/url-state';
+import { formatMilliseconds } from '../../utils/format-numbers';
+
+import type {
+  ConnectedProps,
+  ExplicitConnectOptions,
+} from '../../utils/connect';
+import type { ThreadIndex } from '../../types/profile';
+import type {
+  IndexIntoTracingMarkers,
+  TracingMarker,
+} from '../../types/profile-derived';
+
+type CanCopyContentProps = {|
+  +tagName?: string,
+  +content: string,
+  +className?: string,
+|};
+
+class CanSelectContent extends React.PureComponent<CanCopyContentProps> {
```

Comment:
I think this class could be extracted and shared between the sidebar implementations.